src/Shop.java

```diff
@@ -90,26 +199,48 @@ public Employee FindEmployeeAvailable(String skillNeeded) {
         return null;
     }
 
-    public boolean isProductAvailable(String name) {
-        for (int i = 0; i < products.size(); i++) {
-            if (products.get(i).getProduct().getName() == name && products.get(i).getQuantity() > 0) {
-                return true;
-            }
+    public int getNumberOfEmployees() {
+        int counter = 0;
+        for (Employee employee: employees) {
+            counter++;
         }
-        return false;
+        return counter;
     }
 
-    public void buyProduct(String name) {
-        if (isProductAvailable(name)) {
-            for (int i = 0; i < products.size(); i++) {
-                if (products.get(i).getProduct().getName() == name) {
-                    products.get(i).setQuantity(products.get(i).getQuantity() - 1);
-                }
+    // ********** End of Methods & CRUD for Employees ***********
+
+    // ********** Methods - Crud for Services **********
+    // Constructor for Service - String 'name', double 'price', Integer 'duration', String 'skillNeeded'
+    public void addService(Service service) {
+        services.add(service);
+    }
+
+    public Service findServiceById(int id) {
+        for (Service service: services) {
+            if (service.getId() == id) {
+                return service;
             }
         }
+        return null;
+    }
+
+    public void deleteService(int id) {
+        Service toDelete = findServiceById(id);
+        if (toDelete != null) {
+            services.remove(toDelete);
+        }
+    }
+
+    public void updateService(int id, Service service) {
+        Service toUpdate = findServiceById(id);
+        if (toUpdate != null) {
+            services.remove(findServiceById(id));
```

Comment:
instead of `services.remove(findServiceById(id));` you can do `services.remove(toUpdate);`

Reply:
Missed that one! Thanks!